flake-module.nix

```diff
@@ -0,0 +1,81 @@
+{ self, lib, flake-parts-lib, ... }:
+let
+  inherit (flake-parts-lib)
+    mkPerSystemOption;
+  inherit (lib)
+    mkOption
+    types;
+in
+{
+  options = {
+    perSystem = mkPerSystemOption
+      ({ config, self', inputs', pkgs, system, ... }: {
+        options.treefmt = mkOption {
+          description = ''
+            treefmt flake module options
+          '';
+          type = types.submodule {
```

Comment:
do you think it's possible to graft the other module directly into here, so the options can be exposed directly?

Reply:
@zimbatm I've done this now. `config`'s type is now derived from module-options.nix. 

What do you think of the top-level of `wrapper` option? It just a convenient (shorter-to-type) alias to `module.config.build.wrapper` now. As for the others:

- `programs`: We need this to gather all formatters in one place, so they can be put in the devShell's nativeBuildInputs which is needed to get editors and IDEs (running nix environment) recognize the individual formatters.
- `module`: This is not meant to be used by the user generally, it simply allows us to evaluate the module once rather than twice. Though it could be useful in some advanced cases.

Reply:
if `programs` is move to the main module I think that's enough, then we can get rid of the evalModule call below.